From ba0f400e15af85e1521a078e2c256c00ba8872df Mon Sep 17 00:00:00 2001
From: Thomas Lambertz <mail@thomaslambertz.de>
Date: Thu, 23 Apr 2020 19:50:19 +0200
Subject: [PATCH] change clock source to TSC

---
 libmcount/internal.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libmcount/internal.h b/libmcount/internal.h
index 5fadfcaf..a1205084 100644
--- a/libmcount/internal.h
+++ b/libmcount/internal.h
@@ -197,9 +197,9 @@ static inline void mcount_watch_release(struct mcount_thread_data *mtdp) {}
 
 static inline uint64_t mcount_gettime(void)
 {
-	struct timespec ts;
-	clock_gettime(CLOCK_MONOTONIC, &ts);
-	return (uint64_t)ts.tv_sec * NSEC_PER_SEC + ts.tv_nsec;
+	uint32_t a, d;
+	asm volatile("rdtsc" : "=a" (a), "=d" (d));
+	return (((uint64_t)a) | (((uint64_t)d) << 32));
 }
 
 static inline int mcount_gettid(struct mcount_thread_data *mtdp)
-- 
2.26.1

